toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -806,6 +813,57 @@ func validateOutput(baseConfigPath string, output imagecustomizerapi.Output, out
 	return nil
 }
 
+func validateStorage(storage *imagecustomizerapi.Storage) error {
+	if storage == nil {
+		return nil
+	}
+
+	err := validateVeritySignatureInjection(storage.Verity, storage.FileSystems)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateVeritySignatureInjection(verityList []imagecustomizerapi.Verity,
+	fileSystems []imagecustomizerapi.FileSystem,
+) error {
+	if len(verityList) == 0 {
+		return nil
+	}
+
+	var espMount string
+
+	for _, fs := range fileSystems {
+		if fs.Type == "fat32" && fs.MountPoint != nil {
+			espMount = fs.MountPoint.Path
+			break
+		}
+	}
+
+	if espMount == "" {
+		return fmt.Errorf("could not find mount point for ESP partition (fat32)")
+	}
+
+	espMountClean := filepath.Clean(espMount)
+
+	for _, verity := range verityList {
+		if verity.HashSignatureInjection == "" {
+			continue
+		}
+
+		sigPath := filepath.Clean(verity.HashSignatureInjection)
+
+		// Ensure it's under the ESP mount point
+		if !strings.HasPrefix(sigPath, espMountClean+"/") {
```

Comment:
It's worth noting that `Storage.IsValid` has a check that ensures the ESP is mounted at `/boot/efi`.